src/Castle.Facilities.Synchronize/CreateOnUIThreadInspector.cs

```diff
@@ -171,8 +173,11 @@ private static IReference<IProxyGenerationHook> ObtainProxyHook(IConfiguration c
 						var message = String.Format("The specified controlProxyHook does " +
 						                            "not implement the interface {1}. Type {0}",
 						                            hookType.FullName, typeof(IProxyGenerationHook).FullName);
-
+#if FEATURE_SYSTEM_CONFIGURATION
 						throw new ConfigurationErrorsException(message);
+#else
+						throw new InvalidOperationException(message);
+#endif
```

Comment:
You could just change it to `System.Exception` in all cases because it doesn't quite sound like `ConfigurationErrorsException` should have been used. However it is all a bit moot because this whole file needs to be wrapped in `#if FEATURE_WINFORMS`.